### PR TITLE
Removed unused errors

### LINKED ISSUE #645: Audit: Remove unused Errors and other enums
There are several error enums that are unused; review these and understand why they are there, use if we should and delete if not.  This should be done prior to #622

Relates to #255 

## Schemas pallet
- ExceedsGovernanceSchemaModelMaxValue
- NoSuchSchema
- StringConversionError
- DeserializationError
- SerializationError

## Messages pallet
(None found)

## Msa pallet
- MoreThanOneKeyExists
- RegisteredProviderCannotBeRetired

### PR DESCRIPTION
**Pull request by dopatraman**
# Goal
The goal of this PR is to remove unused error enums.

Closes #645

### CODE REVIEW

**Review by enddynayn — APPROVED** [review]
Nice! 💯

File: pallets/msa/src/lib.rs [hunk 1]
```diff
@@ -300,12 +300,6 @@ pub mod pallet {
 		/// The number of key values has reached its maximum
 		KeyLimitExceeded,
 
-		/// More than one account key exists for the MSA during retire attempt
-		MoreThanOneKeyExists,
-
-		/// Can't retire a registered provider MSA
-		RegisteredProviderCannotBeRetired,
```
Comment (shannonwells) [inline pallets/msa/src/lib.rs:307]: This error is now part of the CheckFreeExtrinsic use ValidityErrors.

File: pallets/schemas/src/lib.rs [hunk 2]
```diff
@@ -122,24 +122,9 @@ pub mod pallet {
...
 		ExceedsMaxSchemaModelBytes,
 
-		/// The governance schema model max value provided is too large (greater than the BoundedVec size)
-		ExceedsGovernanceSchemaModelMaxValue,
-
 		/// The schema is less than the minimum length allowed
 		LessThanMinSchemaModelBytes,
 
-		/// Schema does not exist
-		NoSuchSchema,
```
Comment (shannonwells) [inline pallets/schemas/src/lib.rs:132]: It would be nice to know what happened to these errors; were they ever in use?

**Review by shannonwells — APPROVED** [review]

File: pallets/msa/src/lib.rs (hunk 1 above)
Comment (aramikm) [reply pallets/msa/src/lib.rs:307]: do we need to tag people from Sas team for any change that might break their end or do we have a process for this?

**Review by aramikm — APPROVED** [review]

File: pallets/schemas/src/lib.rs (hunk 2 above)
Comment (dopatraman) [reply pallets/schemas/src/lib.rs:132]: I wonder if this particular error was replaced by `InvalidSchemaId` in the messages pallet:
```
let schema = T::SchemaProvider::get_schema_by_id(schema_id);
			ensure!(schema.is_some(), Error::<T>::InvalidSchemaId);
```

**Review by saraswatpuneet — APPROVED** [review]

**Pull request merged**